### PR TITLE
Allow MatrixAggregatable to be registered as an (optional) extension

### PR DESCRIPTION
**Pull request by jglick**
This allows you to stop using `implements MatrixAggregatable` on publishers, which forces a hard dep on this plugin, and start using an optional extension.

@reviewbybees

### CODE REVIEW

File: src/main/java/hudson/matrix/MatrixAggregatable.java [hunk 1]
```diff
@@ -33,7 +34,7 @@
 /**
  * {@link Publisher}, {@link JobProperty}, {@link BuildWrapper} can optionally implement this interface
  * to perform result aggregation across {@link MatrixRun}.
- *
+ * <p>You may also register singleton instances of this as {@link Extension}s.
```
Comment (olivergondza) [inline src/main/java/hudson/matrix/MatrixAggregatable.java]: This is a great improvement but I am not sure I would understand this from the comment.

If I am not mistaken, these would actually be "global" aggregatos as they will not be configured in job (unlike Publisher, JobProperty, or BuildWrapper). I find this worth mentioning.

**Comment by olivergondza** [comment]
👍 with minor doc comment.

**Comment by jglick** [comment]
@reviewbybees done

**Comment by daniel-beck** [comment]
Isn't it time this gets released?

**Comment by olivergondza** [comment]
@daniel-beck coincidentally working on another fix in the plugin so will try to do it today.

**Pull request merged**